tests/Application/Actions/Hooks/StripePaymentsUpdateTest.php

```diff
@@ -142,12 +139,7 @@ public function testSuccessfulPayment(): void
         $body = json_encode($webhookContent);
         $webhookSecret = $this->getValidWebhookSecret($container);
         $time = (string) time();
-
-        $donationRepoProphecy = $this->prophesize(DonationRepository::class);
-        $donationRepoProphecy
-            ->findAndLockOneBy(['transactionId' => 'pi_externalId_123'])
-            ->willReturn($donation)
-            ->shouldBeCalledOnce();
+        $this->donationRepository->store($donation);
```

Comment:
Liking this. Seems like right now, this is the main win from the extra layer of abstraction having the repo as an interface, but the expectation is there will be a longer term gain as we implement more mocked methods in the in-memory version for more tests?

Reply:
Yes I think in many tests it will be useful to have an in memory repo and using that should make it a lot easier to change the repo api since the many tests don't each constrain it directly.

But it is still a bit of an awkward design since there are still many non persistence related functions in the repo.